syntax = "proto3";

package users;

message User {
    int32 id = 1;
    string username = 2;
    string full_name = 3;
    string email = 4;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    bool success = 1;
    User user = 2;
}

message UpdateUserRequest {
    int32 id = 1;
    string full_name = 2;
    string email = 3;
}

service UserService {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc UpdateUser(UpdateUserRequest) returns (User);
}
